### PR TITLE
get list token

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
image

File: src/main/java/microservices/token/steps/TokenSteps.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import constants.Endpoints;
+import core.BaseApi;
+import io.qameta.allure.Step;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TokenSteps extends BaseApi {
+    @Step
```
Comment (Sotatek-DavidNguyen4) [inline src/main/java/microservices/token/steps/TokenSteps.java]: add description for this step

File: src/main/java/microservices/token/steps/TokenSteps.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public String getListTokensWithPageInvalid(String page, int size, ArrayList<String> sort){
+        Map<String, Object> paramsToken = new HashMap<>();
+        paramsToken.put("page", page);
+        paramsToken.put("size", size);
+        paramsToken.put("sort", sort.get(0));
+        paramsToken.put("sort", sort.get(1));
+        sendGet(Endpoints.TokenApi.GET_LIST_TOKEN, paramsToken);
+        return getJsonAsString();
+    }
+    @Step
```
Comment (Sotatek-DavidNguyen4) [inline src/main/java/microservices/token/steps/TokenSteps.java]: here too

File: src/main/java/microservices/token/steps/TokenSteps.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    public String getListTokensWithSizeInvalid(int page, String size, ArrayList<String> sort){
+        Map<String, Object> paramsToken = new HashMap<>();
+        paramsToken.put("page", page);
+        paramsToken.put("size", size);
+        paramsToken.put("sort", sort.get(0));
+        paramsToken.put("sort", sort.get(1));
+        sendGet(Endpoints.TokenApi.GET_LIST_TOKEN, paramsToken);
+        return getJsonAsString();
+    }
+    @Step
```
Comment (Sotatek-DavidNguyen4) [inline src/main/java/microservices/token/steps/TokenSteps.java]: here too

File: src/test/java/tests/TokenTests.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    private Token token = new Token();
+    BaseApi baseApi = new BaseApi();
+    @Test(description = "verify that get list token successfull", groups={"token"}, dataProvider = "paramSuccess")
+    public void getListTokenSuccess(int page, int size, ArrayList<String> sort){
+        TokenSteps tokenSteps = new TokenSteps();
+        tokenSteps.getListTokens(page, size, sort);
+        token = (Token) tokenSteps.saveResponseObject(Token.class);
+        ArrayList<DataToken> dataToken = token.getData();
+        baseApi.validateResponse(200);
+        for(int i =1; i <dataToken.size(); i++){
```
Comment (Sotatek-DavidNguyen4) [inline src/test/java/tests/TokenTests.java]: put in another function

Comment (ghost) [reply src/test/java/tests/TokenTests.java]: okay bro

**Comment by ghost** [comment]
image

**Review by Sotatek-DavidNguyen4 — APPROVED** [review]

**Pull request merged**